[3.13] gh-133745: Fix asyncio task factory name/context kwarg breaks (#133948)
python/cpython: Doc/library/asyncio-eventloop.rst

   .. versionadded:: 3.5.2

> .. method:: loop.create_task(coro, *, name=None, context=None, eager_start=None, **kwargs)

Author: ```suggestion
.. method:: loop.create_task(coro, *, name=None, context=None, **kwargs)
```

   for interoperability. In this case, the result type is a subclass
   of :class:`Task`.

   The full function signature is largely the same as that of the
   :class:`Task` constructor (or factory) - all of the keyword arguments to
   this function are passed through to that interface, except *name*,
   or *context* if it is ``None``.

   If the *name* argument is provided and not ``None``, it is set as
   the name of the task using :meth:`Task.set_name`.

   An optional keyword-only *context* argument allows specifying a
   custom :class:`contextvars.Context` for the *coro* to run in.
   The current context copy is created when no *context* is provided.

>    An optional keyword-only *eager_start* argument allows eagerly starting
>    the execution of the :class:`asyncio.Task` at task creation time.
>    If set to ``True`` and the event loop is running, the task will start
>    executing the coroutine immediately, until the first time the coroutine
>    blocks. If the coroutine returns or raises without blocking, the task
>    will be finished eagerly and will skip scheduling to the event loop.
> 

Author: ```suggestion
```

   for interoperability. In this case, the result type is a subclass
   of :class:`Task`.

   The full function signature is largely the same as that of the
   :class:`Task` constructor (or factory) - all of the keyword arguments to
   this function are passed through to that interface, except *name*,
   or *context* if it is ``None``.

   If the *name* argument is provided and not ``None``, it is set as
   the name of the task using :meth:`Task.set_name`.

   An optional keyword-only *context* argument allows specifying a
   custom :class:`contextvars.Context` for the *coro* to run in.
   The current context copy is created when no *context* is provided.

   An optional keyword-only *eager_start* argument allows eagerly starting
   the execution of the :class:`asyncio.Task` at task creation time.
   If set to ``True`` and the event loop is running, the task will start
   executing the coroutine immediately, until the first time the coroutine
   blocks. If the coroutine returns or raises without blocking, the task
   will be finished eagerly and will skip scheduling to the event loop.

> 

Reviewer: I don't believe we should document `eager_start` here at all. I doesn't require special treatment, it can just be put in `**kwargs`. It looks like we added this in 3.13.4, and such feature changes in a bugfix release are frowned upon. We should only advertise `**kwargs` as an unfortunate added in3.13.3 and fixed in 3.13.4.

   .. versionchanged:: 3.8
      Added the *name* parameter.

   .. versionchanged:: 3.11
      Added the *context* parameter.

   .. versionchanged:: 3.13.3

>       Added ``kwargs`` which always passes on ``kwargs`` such as the *eager_start*
>       parameter and *name* parameter.

Reviewer: Again, let's not advertise support for eager_tasks here, as it's not in 3.13.[0-2].

```suggestion
      Added ``kwargs`` which passes on arbitrary extra parameters, including  ``name`` and ``context``.
```

   .. versionchanged:: 3.13.4
      Rolled back the change that passes on *name* and *context* (if it is None),

>       passing on new keword arguments such as *eager_start* is still supported.

Reviewer 2: ```suggestion
      passing on new keyword arguments such as *eager_start* is still supported.
```

>       passing on new keyword arguments such as *eager_start* is still supported.

Reviewer: ```suggestion
      while still passing on other arbitrary keyword arguments (to avoid breaking backwards compatibility with 3.13.3).
```

   for interoperability. In this case, the result type is a subclass
   of :class:`Task`.

>    The arguments shown above are merely the most common ones, described below

Author: ```suggestion
```

   event loop, and *coro* is a coroutine object.  The callable
   must pass on all *kwargs*, and return a :class:`asyncio.Task`-compatible object.

>    .. versionchanged:: 3.13.3
>    Required that all *kwargs* are passed on to :class:`asyncio.Task`.

Reviewer: ```suggestion
   .. versionchanged:: 3.13.3
      Required that all *kwargs* are passed on to :class:`asyncio.Task`.
```

>    .. versionchanged:: 3.13.4
>    *name* is no longer passed to task factories. *context* is no longer passed
>    to task factories if it is ``None``.

Reviewer: ```suggestion
   .. versionchanged:: 3.13.4
      *name* is no longer passed to task factories. *context* is no longer passed
      to task factories if it is ``None``.
```
